composeApp/src/commonMain/kotlin/io/github/drp08/studypal/screens/RegistrationScreen.kt

```diff
@@ -0,0 +1,168 @@
+                    text = "and",
+                    modifier = Modifier.padding(horizontal = 8.dp),
+                    style = MaterialTheme.typography.body1
+                )
+
+                TextField(
+                    value = viewController.workingHoursEnd,
+                    onValueChange = viewController::onWorkingHoursEndChange,
+                    label = { Text("End") },
+                    modifier = Modifier
+                        .weight(1f)
+                        .padding(start = 8.dp),
+                    singleLine = true,
+                    shape = RoundedCornerShape(8.dp),
+                    colors = TextFieldDefaults.textFieldColors(
+                        backgroundColor = MaterialTheme.colors.surface
+                    ),
+                    keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number)
+                )
+            }
+
+            TextField(
+                value = viewController.hoursPerDay,
+                onValueChange = viewController::onHoursPerDayChange,
+                label = { Text("I want to work ... Hours per day") },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(vertical = 8.dp),
+                singleLine = true,
+                shape = RoundedCornerShape(8.dp),
+                colors = TextFieldDefaults.textFieldColors(
+                    backgroundColor = MaterialTheme.colors.surface
+                ),
+                keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number)
+            )
+
+            Spacer(modifier = Modifier.height(24.dp))
+
+            Button(
+                onClick = { /* Handle Google Calendar import */ },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(50.dp),
+                shape = RoundedCornerShape(8.dp)
+            ) {
+                Text(text = "Import Calendar", fontSize = 18.sp)
+            }
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Button(
+                onClick = { /* Handle registration */ },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(50.dp),
+                shape = RoundedCornerShape(8.dp)
+            ) {
+                Text(text = "Register", fontSize = 18.sp)
+            }
+        }
+    }
+}
+
+class ViewController {
+    var name by mutableStateOf("")
+    var workingHoursStart by mutableStateOf("")
+    var workingHoursEnd by mutableStateOf("")
+    var hoursPerDay by mutableStateOf("")
+
+    fun onNameChange(newValue: String) {
+        name = newValue
+    }
+
+    fun onWorkingHoursStartChange(newValue: String) {
+        workingHoursStart = newValue
+    }
+
+    fun onWorkingHoursEndChange(newValue: String) {
+        workingHoursEnd = newValue
+    }
+
+    fun onHoursPerDayChange(newValue: String) {
+        hoursPerDay = newValue
+    }
+}
```

Comment:
This should be analogous to `HomeScreen` and `HomeViewModel`. Have a single data class inside a `SharedFlow` and collect that result in the composable function.